source/opt/instruction.h

```diff
@@ -347,6 +347,8 @@ class Instruction : public utils::IntrusiveNodeBase<Instruction> {
   Instruction* InsertBefore(std::unique_ptr<Instruction>&& i);
   using utils::IntrusiveNodeBase<Instruction>::InsertBefore;
 
+  inline bool IsConstant() const;
```

Comment:
Should have a method comment. 
You want to include OpContstant* but exclude OpSpecConstant*

(The SPIR-V spec includes spec constant instructions in "constant instructions", so it's best to be crystal clear here.)

Reply:
Done.